synapse_rock/rockcraft.yaml

```diff
@@ -10,176 +10,219 @@ build-base: ubuntu@24.04
+      cd synapse-s3-storage-provider
+      pip3 install --prefix="/install" --no-warn-script-location -U .
+      cd ..
+      cp -r synapse /synapse/
+      cp -r rust /synapse/
+      cp pyproject.toml README.rst build_rust.py Cargo.toml Cargo.lock /synapse/
+      pip3 install --prefix="/install" --no-deps --no-warn-script-location /synapse[all];
+      git clone https://git.buechner.me/nbuechner/synapse-invite-checker
+      cd synapse-invite-checker
+      git checkout ab6c8b78d78c4cbf31e1a30981ae45c09285b34a
+      pip3 install --prefix="/install" --no-warn-script-location -U .
+      cd ..
+      # fix issue while creating file
+      # https://github.com/element-hq/synapse/issues/17882
+      pip3 install --prefix="/install" --force-reinstall -v "Twisted==24.7.0"
+      cp docker/start.py $CRAFT_PART_INSTALL/
+      chmod 755 $CRAFT_PART_INSTALL/start.py
+      cp -r docker/conf $CRAFT_PART_INSTALL/
+      cp -r /usr/local $CRAFT_PART_INSTALL/usr/
+      cp -r /install/local/* $CRAFT_PART_INSTALL/usr/local/
+      mkdir -p $CRAFT_PART_INSTALL/usr/local/attributemaps
+      chmod 755 $CRAFT_PART_INSTALL/usr/local/attributemaps
+      rm -rf $CRAFT_PART_INSTALL/aws/dist/awscli/examples/
+    overlay-packages:
+      - ca-certificates
+      - libjemalloc2
+  patches:
+    plugin: dump
+    after: [synapse]
+    source: patches
+    organize:
+      "*": srv/patches/
+  patch-start:
+    after:
+      - patches
+    plugin: nil
+    build-packages:
+      - git
+    override-stage: |
+      patch -f -u ./start.py -i srv/patches/start.patch
+  synapse-conf:
+    plugin: dump
+    source: attributemaps
+    organize:
+      login_ubuntu.py: /usr/local/attributemaps/login_ubuntu.py
+  mas-templates:
+    plugin: dump
+    <<: *mas-source
+    source-subdir: templates
+    organize:
+      "*": mas/share/templates/
+  mas-translations:
+    plugin: dump
+    <<: *mas-source
+    source-subdir: translations
+    organize:
+      "*": mas/share/translations/
+  mas-assets:
+    plugin: nil
+    <<: *mas-source
+    source-subdir: frontend
+    build-environment:
+      - NODE_URI: "https://nodejs.org/dist/v22.11.0/node-v22.11.0-linux-x64.tar.gz"
+      - NODE_ENV: dev
+    override-build: |
+      curl -Ls $NODE_URI | tar xzf - -C /usr/ --skip-old-files --no-same-owner --strip-components=1
+      npm --prefix=$CRAFT_PART_BUILD/frontend ci
+      npm --prefix=$CRAFT_PART_BUILD/frontend run build
+      mkdir -p $CRAFT_PART_INSTALL/mas/share/assets
+      cp -r frontend/dist/* $CRAFT_PART_INSTALL/mas/share/assets/
+      cp frontend/dist/manifest.json $CRAFT_PART_INSTALL/mas/share/manifest.json
+    stage:
+      - mas/*
+  mas-cli:
+    plugin: rust
+    rust-channel: stable
+    rust-cargo-parameters:
+      - "--features dist"
+      - "--no-default-features"
+    <<: *mas-source
+    override-build: |
+      curl -L -o opa https://openpolicyagent.org/downloads/v0.70.0/opa_linux_amd64_static
```

Comment:
They are using a different version in the upstream. Could this make a difference during execution?
https://github.com/element-hq/matrix-authentication-service/blob/main/Dockerfile#L13

Reply:
(same for rust)

Reply:
hmm, I'm not sure if we need to match the version exactly since they're build time dependencies rather than runtime

Reply:
I think would be safer using same versions to avoid surprises in the future

Reply:
Makes sense, but if you look at the [installation guide](https://element-hq.github.io/matrix-authentication-service/setup/installation.html#building-from-the-source) for MAS, it doesn't actually pin the exact version of OPA or npm, while the rust toolchain should be on `stable` which is currently being used in the rock already.

Reply:
So I think we don't need to match what's in the dockerfile, we're building with `--feature dist` instead of `--feature docker` anyways.

Reply:
I see, makes sense.